rate.go

```diff
@@ -113,6 +113,9 @@ func PerHour(n int) Rate { return Rate{time.Hour / time.Duration(n), n} }
 // PerDay represents a number of requests per day.
 func PerDay(n int) Rate { return Rate{24 * time.Hour / time.Duration(n), n} }
 
+// PerPeriod represents a number of requests per provided period.
+func PerPeriod(n int, period time.Duration) Rate { return Rate{period/ time.Duration(n), n}}
```

Comment:
Do you think we could rename this to `PerDuration`? For two reasons:

1. The "period" that we talk about in this function is different than the `period` that we see in `Rate`. The one in `Rate` is the final computed period that we get by dividing a duration by a count, while the one in this function is the duration that we plug into one end of that division.
2. "Duration" is the name of the construct in Go (i.e. `time.Duration`), so I think that'll just be easier for people to find.

Reply:
I think you'll need to run Gofmt on this (there's a space on only one side of the `/`).

Our build wasn't previously checking for that properly, but I've since fixed it up so that it does.